src/features/Folding.ts

```diff
@@ -421,9 +421,9 @@ export class FoldingProvider implements vscode.FoldingRangeProvider {
     ): ITokenList {
         const result = [];
 
-        const emptyLine = /^[\s]+$/;
-        const startRegionText = /^#\s*region\b/;
-        const endRegionText = /^#\s*endregion\b/;
+        const emptyLine = /^\s*$/;
+        const startRegionText = /^#region\b/i;
```

Comment:
Wait, I think you were right about the whitespace here -- no? That the regex should look like `/^\s*#region/i`?

Reply:
As in, like the other VSCode folding regexes

Reply:
Nope....the regex I use is slightly different, because it's checking the grammar token (it's within the line) not the whole line.  That's why I do an Empty Line check.

Reply:
In the [code below it](https://github.com/glennsarti/vscode-powershell/blob/c61ca6d7561cb13e23398b8cd17a457f9ab58d8c/src/features/Folding.ts#L431-L432), it looks like `subsequentText()` gets the whole comment line and then tests it against the regex, which starts with `^`. Which I read as not matching an indented `#region`. Does that sound right?

Reply:
So the logic I'm using is;

```
    Find the `#` character (This is the `punctuation.definition.comment.powershell` scope)
        If there is only whitespace before the `#` then
            if the text after, and including the `#` matches the starting regex then
                #Do Something
            end
            if the text after, and including the `#` matches the ending regex then
                #Do Something
          end
       end
     end
```

That said, it could be optimized by grabbing the whole line and regex-ing on that...but that's for another PR.

Reply:
Oh, I see.  I think we want `/^\s*#region\b/i`, right?  We don't want variable # spaces between the `#` and `region|endregion` but we do want to allow that *before* the `#`.

Reply:
Also, looks like the `subsequentText()` named is wrong.  Should be `textFrom` or something like that.

Reply:
Oh, so it does this:
```
    #   Some text is here
    ^                   ^
    [offset]            [end of line]
```

`subsequentTest()` seems as good a name as any - but maybe the comment should be something like:
> Get the text from a given offset to the end of the line it's on

Reply:
yup

Reply:
Anyway, I'm satisfied!

Reply:
@rjmholt PR https://github.com/PowerShell/vscode-powershell/pull/1438 refactors the detection to be easier to understand.